File: app/src/main/java/com/example/petridelivery/lists/adapters/abs/PagingColumnHeaderDataAdapter.java
package com.example.petridelivery.lists.adapters.abs;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.paging.PagingDataAdapter;
import androidx.recyclerview.widget.DiffUtil;

import org.jetbrains.annotations.NotNull;

public abstract class PagingColumnHeaderDataAdapter<T, U extends BaseViewHolder> extends PagingDataAdapter<T, U> {
    final static int HEADER = 0;
    final static int ITEM = 1;

    protected LayoutInflater inflater;
    protected ViewGroup parent;
    protected View view;

    public PagingColumnHeaderDataAdapter(@NotNull DiffUtil.ItemCallback<T> diffCallback, View view) {
        super(diffCallback);
        this.view = view;
    }

    @NonNull
    @Override
    public U onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        this.parent = parent;
        inflater = LayoutInflater.from(parent.getContext());

        U viewHolder = createViewHolder(parent);
        if(viewType == HEADER){
            viewHolder.getActionButtonFrame().removeAllViews();
        }
        return viewHolder;
    }

    public abstract U createViewHolder(ViewGroup parent);

    @Override
    public int getItemViewType(int position) {
        if(position == 0){
            return HEADER;
        }
        return ITEM;
    }

    @Override
    public int getItemCount() {
        return super.getItemCount() + 1;
    }
}

File: app/src/main/java/com/example/petridelivery/lists/adapters/abs/PagingColumnHeaderDataAdapter.kt
package com.example.petridelivery.lists.adapters.abs

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.paging.PagingDataAdapter
import androidx.recyclerview.widget.DiffUtil

abstract class PagingColumnHeaderDataAdapter<T: Any, U : EditableViewHolder>(diffCallback: DiffUtil.ItemCallback<T>, protected var view: View) : PagingDataAdapter<T, U>(diffCallback) {
    protected lateinit var inflater: LayoutInflater
    protected lateinit var parent: ViewGroup

    companion object {
        const val HEADER = 0
        const val ITEM = 1
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): U {
        this.parent = parent
        inflater = LayoutInflater.from(parent.context)

        val viewHolder = createViewHolder(parent)
        if (viewType == HEADER) {
            viewHolder.actionButtonFrame.removeAllViews()
        }
        return viewHolder
    }

    abstract fun createViewHolder(parent: ViewGroup): U

    override fun getItemViewType(position: Int): Int {
        return if (position == 0) {
            HEADER
        } else ITEM
    }

    override fun getItemCount(): Int {
        return super.getItemCount() + 1
    }

}